src/main/java/net/glowstone/util/nbt/CompoundTag.java

```diff
@@ -59,6 +59,38 @@ public void remove(String key) {
         value.remove(key);
     }
 
+    /**
+     * Checks to see if this tag is a strict, deep submap of the given CompoundTag.
+     * @param other The CompoundTag that should contain our values.
+     * @return
+     */
+    public boolean matches(CompoundTag other) {
```

Comment:
After testing the following command in my world with only 1 saddled pig: `/testfor @e[type=pig] {Saddle:1b}`, it still matched with pigs that weren't saddled. Either there is a problem with the logic in this function or with the way it is called in the command.

It would be a good idea to have a simple unit test to check if two compounds match using this function.